### ISSUE TITLE
How can i get current progress value?

### ISSUE THREAD

**Issue by niklasgrewe** [kindoflew/svelte-parallax#16]
Hi, thank you very much for this awesome library. 
Recently i saw this apple clone website using webflow: https://apple-october-2020.webflow.io/

I want to build a similiar landing page with `svelte-parallax`, but how can i do that? I noticed some opacity animations, when i section gets visible. It would be nice, if we can bind to some progress/scroll value to add this kind of animations as well. Is that possible?

Could you maybe make a similar demo so i can see it in action? 
Many thanks

**Comment by kindoflew** [comment]
Hi, thanks for the kind words!

I'm currently working on a feature that would surface scroll progress in some way (my current idea is an `onProgress` prop that would be a function that receives the current scroll progress).

In the meantime, you may be able to use [Intersection Observer](https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API), but I'm not exactly sure how that API interacts with absolutely positioned elements (which the `ParallaxLayer`s are). 

Also, the `Parallax` component just uses Svelte's `scrollY` and `innerHeight` bindings on `window`, so you could use those to figure out how far you've scrolled manually:

```html
<script>
  // bind:scrollY
  let scrollY;
  // bind:innerHeight
  let innerHeight;
  let parallaxHeight = innerHeight * numberOfParallaxSections;
  $: progress = (scrollY / parallaxHeight) * 100;
</script>

<svelte:window bind:scrollY bind:innerHeight />
```
This is assuming `Parallax` is the top-level element on the page, or you are using some kind of `clamp` function to keep the value of `progress` between `0` and `100`.

You could also get the current `section` you are in and its scroll progress with something like this:

```html
<script>
  // bind:scrollY
  let scrollY;
  // bind:innerHeight
  let innerHeight;
  $: currentSection = Math.floor((scrollY / innerHeight) * numberOfParallaxSections) + 1; // add 1 because it's zero-indexed
  $: currentSectionProgress = (scrollY / ((innerHeight * (currentSection)) * 100;
</script>

<svelte:window bind:scrollY bind:innerHeight />
```

A little boilerplate-y, which is one of the reasons why I'm trying to figure out good ways to do this out of the box. 

I'm going to keep this open until these features are finished and added. I don't really have a timeframe, because I do this in my free time, but I can ping you here when it's done!

**Comment by niklasgrewe** [comment]
thank you very much for your detailed answer. Glad to hear that you are planning something in that direction with `onProgress`. Until then, I'll use your code examples, thanks.

I still have one question. As I mentioned before, I would like to recreate the parallax effect from the Webflow site. However, I am having some struggle with `offset`, `span` and `rate`. Would you have an idea what values I would have to use to get a similar result?

**Comment by kindoflew** [comment]
Sure!

for `span` I would probably leave it at the default `1` for all of them, since they seem to take up the whole viewport. `span` is just how many viewport-heights tall the `ParallaxLayer` is.

for `offset` i would set each one to the top of which `section` i'd want it to be at (so integers from `0`-`4`, assuming `5` sections -- one for each `ParallaxLayer`). `offset` is basically where you want your `ParallaxLayer` to be when it is scrolled to, in relation to the top the `Parallax` container (so, for example, if you want a `ParallaxLayer` to take up the whole screen after you have scrolled down one `section`, it's `offset` should be `1` -- it will be `1` viewport-height from the top).

for `rate` it would kind of depend on how fast you want them to move relative to the user's scroll, but the default value of `0.5` seems like it might be about the right speed to match your Webflow example. i probably wouldn't set it higher than `1` though.

**Comment by kindoflew** [comment]
hey @niklasgrewe! this feature is now available in `v0.4.0`! let me know if anything isn't working! thanks for this request!
